src/adapter/src/catalog.rs

```diff
@@ -1008,6 +1017,30 @@ impl CatalogState {
         assert!(cluster.replicas_by_id.insert(replica_id, replica).is_none());
     }
 
+    fn rename_cluster_replica(
+        &mut self,
+        cluster_id: ClusterId,
+        replica_id: ReplicaId,
+        to_name: String,
+    ) {
+        let cluster = self
+            .clusters_by_id
+            .get_mut(&cluster_id)
+            .expect("Missing cluster");
+        let replica = cluster
+            .replicas_by_id
+            .get_mut(&replica_id)
+            .expect("Missing cluster");
```

Comment:
```suggestion
        let replica = self.get_cluster_replica_mut(cluster_id, replica_id);
```

You'll need to add this helper, but I think it's worth it if you do so with the same panic message that `get_cluster_replica` uses. We have every other variant: `try_get_cluster_replica`, `get_cluster_replica`, and `try_get_cluster_replica_mut`.

Reply:
Thanks, fixed.